test/util.js

```diff
@@ -513,3 +515,261 @@ function checkCursorPaginatedHistoryObject ({
     }
   })
 }
+
+/**
+ * @param {Object}   params
+ * @param {Object}   params.t - AVA test object
+ * @param {Object}   params.authorizationHeaders
+ * @param {String}   params.endpointUrl - tested endpoint
+ * @param {String}   [params.fetchEndpointUrl] - if not specified, fallback to endpointUrl
+ *   is the endpoint to hit to auto-generate test values
+ *   (useful to specify it for stats and history endpoints)
+ * @param {Function} [params.checkPaginationObject(t, obj)]
+ * @param {Object[]} params.filters
+ * @param {String}   params.filters[i].prop
+ * @param {Function} [params.filters[i].customGetValue(obj)]
+ *   custom getter to value with prop (by default obj[prop])
+ * @param {Function} [params.filters[i].customCheck(obj, value)]
+ *   if specified, will perform a custom check on that property
+ * @param {Boolean}  [params.filters[i].isArrayFilter = false]
+ * @param {Boolean}  [params.filters[i].isRangeFilter = false]
+ * @param {Object[]} [params.filters[i].customTestValues] - if specified, will only use those values
+ *   and test each one of those values
+ *   otherwise, will perform a fetch of objects and test with fixtures values
+ * @param {Boolean}  [params.filters[i].noResultsExistenceCheck = false]
+ *   by default, will check if there are results while it should have (based on auto-fetched values)
+ */
+async function checkFilters ({
+  t,
+  authorizationHeaders,
+  endpointUrl,
+  fetchEndpointUrl,
+  checkPaginationObject,
+  filters,
+}) {
+  const getEndpoint = (endpointUrl, queryParams) => {
+    if (endpointUrl.includes('?')) return `${endpointUrl}&${queryParams}`
+    else return `${endpointUrl}?${queryParams}`
+  }
+
+  const getRandomValue = (values) => _.first(_.shuffle(values))
+
+  const sortValues = (values) => {
+    return values.slice(0).sort((a, b) => {
+      if (a < b) return -1
+      else if (a > b) return 1
+      else return 0
+    })
+  }
+
+  // prop=value
+  const getExactValueFilter = (prop, exactValue) => `${prop}=${exactValue}`
+
+  // prop[]=value1&prop[]=value2
+  const getArrayValuesFilter = (prop, values) => values.map(v => `${prop}[]=${v}`).join('&')
+
+  // prop=value1,value2
+  const getArrayCommaSeparatedValuesFilter = (prop, values) => `${prop}=${values.join(',')}`
+
+  // prop[gte]=min&prop[lte]=max
+  const getRangeFilter = (prop, range) => `${prop}[gte]=${range.gte}&${prop}[lte]=${range.lte}`
+
+  const addCheckObjFunctions = (filter) => {
+    const {
+      prop,
+      customGetValue,
+      isArrayFilter,
+      isRangeFilter,
+      customCheck
+    } = filter
+
+    const getValue = (obj) => _.isFunction(customGetValue) ? customGetValue(obj) : obj[prop]
+    filter.getValue = getValue
+
+    filter.getExactValueCheckObj = _.curry((exactValue, obj) => {
```

Comment:
It seems to be `get…CheckFn` rather than `get…CheckObj`.

Same for the other functions below.

Reply:
Yes thanks!